code/object/objectdock.cpp

```diff
@@ -108,6 +108,11 @@ bool dock_check_find_docked_object(object *objp, object *other_objp)
 	Assert(objp != NULL);
 	Assert(other_objp != NULL);
 
+	if (!(objp != nullptr && objp->signature > 0))
```

Comment:
Please also add `Assertion`s for the added condition here to catch this in debug builds as well.